### PR TITLE
Compute gcc version automatically from installed RTools

### PR DESCRIPTION
**Pull request by jgutman**
### Intent

Addresses #12382 
### Approach
While we could choose to update RCompilationDatabase.cpp so that `discoverSystemIncludePaths` runs on Windows as well as Unix, it is more straightforward and requires less path/separator normalization to just update the existing RToolsInfo.cpp code so that rather than hardcoding gcc for a particular version of RTools, we infer the gcc version from the installation directory.

### Automated Tests

None
### QA Notes

See https://github.com/rstudio/rstudio/issues/12382#issuecomment-1344583816
### Checklist

- [ ] If this PR adds a new feature, or fixes a bug in a previously released version, it includes an entry in `NEWS.md` 
- [ ] If this PR adds or changes UI, the updated UI meets [accessibility standards](https://github.com/rstudio/rstudio/wiki/Accessibility)
- [X] A reviewer is assigned to this PR (if unsure who to assign, check Area Owners list)
- [X] This PR passes all local unit tests

### CODE REVIEW

File: src/cpp/core/r_util/RToolsInfo.cpp [hunk 1]
```diff
@@ -276,34 +276,50 @@ RToolsInfo::RToolsInfo(const std::string& name,
...
-         "x86_64-w64-mingw32.static.posix/lib/gcc/x86_64-w64-mingw32.static.posix/10.3.0/include/c++/x86_64-w64-mingw32.static.posix",
-         "x86_64-w64-mingw32.static.posix/lib/gcc/x86_64-w64-mingw32.static.posix/10.3.0/include/c++/backward",
-         "x86_64-w64-mingw32.static.posix/lib/gcc/x86_64-w64-mingw32.static.posix/10.3.0/include",
-         "x86_64-w64-mingw32.static.posix/include",
-         "x86_64-w64-mingw32.static.posix/lib/gcc/x86_64-w64-mingw32.static.posix/10.3.0/include-fixed",
-      };
-
-      for (auto&& stem : cppStems)
-         cppIncludePaths.push_back(installPath.completeChildPath(stem).getAbsolutePath());
+      std::string toolDir = "x86_64-w64-mingw32.static.posix";
```
Comment (kevinushey) [inline src/cpp/core/r_util/RToolsInfo.cpp:279]: We don't have any guarantee this directory will still be used in future builds of Rtools. (what if future builds aren't static? what if MinGW changes the name for some reason?)

File: src/cpp/core/r_util/RToolsInfo.cpp [hunk 2]
```diff
@@ -276,34 +276,50 @@ RToolsInfo::RToolsInfo(const std::string& name,
...
+      FilePath gccRootPath = installPath.completeChildPath(toolDir + "/lib/gcc/" + toolDir);
```
Comment (kevinushey) [inline src/cpp/core/r_util/RToolsInfo.cpp:280]: Similarly, I don't think we can assume this directory layout.

**Review by kevinushey — COMMENTED** [review]
I feel pretty strongly that we should infer the compiler flags directly from the output of the gcc / g++ compilers, rather than continuing to try to hardcode them in this sort of fashion.

**Comment by timtmok** [comment]
I think we can call `g++` or `c++` from `rtools42\x86_64-w64-mingw32.static.posix\bin`.  So calling `g++ -E -x c++ - -v` like it does in the non-Windows code would give the same information.

**Comment by jgutman** [comment]
We could, I tried out calling the existing discoverSystemPaths function, which works correctly it just doesn't currently get called on Windows. It gets the right path, we just need some fixes for the separators since it ends up looking kind of ugly. It seemed like a bigger change to do that than to just fix the existing rtools method, but if you feel that's more robust in the long run I can do it that way -- I just ended up reverting those changes

**Comment by kevinushey** [comment]
For some context: in the past, we assumed that new Rtools releases would essentially be frozen, so we could just hard-code the required information about the Rtools installation and update RStudio in tandem with R / Rtools releases.

Now that updates to Rtools are (1) decoupled from R releases (and so might appear at any time), and (2) bound to change in apparently unexpected ways, I think the extra effort to ensure we are extra robust is warranted.

This would also (hopefully) make updates for compatibility with future releases of Rtools easier, as we should be able to re-use much of the existing inference tools rather than re-writing it every time a new version of Rtools is released.

**Comment by jgutman** [comment]
Abandoning in favor of #12465

**Pull request closed without merging**